[Issue 5562][docs]Modify schema compatibility check doc (#5757)
apache/pulsar: site2/docs/schema-evolution-compatibility.md

</tr>

<tr>

<td> 

`UNDEFINED`

</td> 

<td> 

> The schema compatibility check strategy in order to Upgrade to 2.5.0 or above for compatibility with pulsar version below 2.5.0.

Reviewer: ```suggestion
* If you upgrade a lower version to Pulsar 2.5.0, then it uses the `UNDEFINED` strategy, at this time, it actually uses the compatibility strategy of the topic.

* If you use Pulsar 2.5.0 and a higher version, then it uses the `UNDEFINED` strategy, at this time, it actually uses the `FULL` compatibility strategy.
```

</td> 

<td> 

> NONE

Reviewer: ```suggestion
It depends
```

</td> 

<td> 

NONE

> </td> 

Reviewer: ```suggestion
It depends
```

<td> 

> NONE

Reviewer: ```suggestion
It depends
```

In some data formats, for example, Avro, you can define fields with default values. Consequently, adding or removing a field with a default value is a fully compatible change.

### UNDEFINED 

<table style="table">

<tr>

<th>
    
Compatibility check strategy

</th>

<th>
    

> Note 

Reviewer: ```suggestion
Description
```

</th>

</tr>

<tr>

<td> 

`UNDEFINED`

</td> 

<td> 

> The schema compatibility check strategy in order to Upgrade to 2.5.0 or above for compatibility with pulsar version below 2.5.0.

Reviewer: ```suggestion
* If you upgrade a lower version to Pulsar 2.5.0, then it uses the `UNDEFINED` strategy, at this time, it actually uses the compatibility strategy of the topic.

* If you use Pulsar 2.5.0 and a higher version, then it uses the `UNDEFINED` strategy, at this time, it actually uses the `FULL` compatibility strategy.
```

</td> 

</tr>

</table>

> 

Reviewer: Add schema verification here. I've re-written some sentences as well.

```suggestion
## Schema verification

When a producer or a consumer tries to connect to a topic, a broker performs some checks to verify a schema.

### Producer

When a producer tries to connect to a topic (suppose ignore the schema auto creation), a broker does the following checks:

* Check if the schema carried by the producer exists in the schema registry or not.

    * If the schema is already registered, then the producer is connected to a broker and produce messages with that schema.

    * If the schema is not registered, then Pulsar verifies if the schema is allowed to be registered based on the configured compatibility check strategy.

### Consumer

When a consumer tries to connect to a topic, a broker checks if a carried schema is compatible with a registered schema based on the configured schema compatibility check strategy.

<table style="table">

<tr>

<th>

Compatibility check strategy

</th>

<th>

Check logic

</th>

</tr>

<tr>

<td> 

`ALWAYS_COMPATIBLE`

</td> 

<td> 

All pass

</td>  

</tr>

<tr>

<td> 

`ALWAYS_INCOMPATIBLE`

</td> 

<td> 

No pass

</td> 

</tr>

<tr>

<td> 

`BACKWARD`

</td> 

<td> 

Can read the last schema

</td> 

</tr>

<tr>

<td> 

`BACKWARD_TRANSITIVE`

</td> 

<td> 

Can read all schemas

</td> 

</tr>

<tr>

<td> 

`FORWARD`

</td> 

<td> 

Can read the last schema

</td> 

</tr>

<tr>

<td> 

`FORWARD_TRANSITIVE`

</td> 

<td> 

Can read the last schema

</td> 

</tr>

<tr>

<td> 

`FULL`

</td> 

<td> 

Can read the last schema

</td> 

</tr>

<tr>

<td> 

`FULL_TRANSITIVE`

</td> 

<td> 

Can read all schemas

</td> 

</tr>

</table>
```

</tr>

</table>

> ## Schema verification for producers

Reviewer: I've moved this part before `order of upgrading clients`, please delete lines 857-1017.
